google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ResourceLoadingModelItem.java

```diff
@@ -16,12 +16,46 @@
 
 package com.google.cloud.tools.intellij.resources;
 
+import org.jetbrains.annotations.NotNull;
+
 import javax.swing.tree.DefaultMutableTreeNode;
 
 /**
- * This model item is shown in the project selector when a resource manager call is outstanding.
+ * This model item is shown when a resource manager call is outstanding.
  */
 class ResourceLoadingModelItem extends DefaultMutableTreeNode {
 
 }
 
+/**
+ * This model item is shown when a resource manager call returns an empty result.
+ */
+class ResourceEmptyModelItem extends DefaultMutableTreeNode {
+
+  private String message;
+
+  public ResourceEmptyModelItem(@NotNull String message) {
+    this.message = message;
+  }
+
+  public String getMessage() {
+    return message;
+  }
+}
+
+/**
+ * This model item represents a node shown when an error occurs attempting to query a resource
+ * manager.
+ */
+class ResourceErrorModelItem extends DefaultMutableTreeNode {
```

Comment:
will leave as is since all these classes are very trivial so I'd prefer to just group them all in the same file (there are 3 now)